src/outline.rs

```diff
@@ -347,11 +347,8 @@ impl<'a> OutlineNodeRef<'a> {
                 let end = if let Some(next_stmt) = self.next_after_children() {
                     next_stmt.get_statement().span().start - 1
                 } else {
-                    database
-                        .parse_result()
-                        .source_info(stmt.segment.id)
-                        .span
-                        .end
+                    let segment_span = database.parse_result().source_info(stmt.segment.id).span;
+                    segment_span.end - segment_span.start
```

Comment:
why is this `let end = end - start`? It doesn't make sense to me.

Reply:
Here there is no next statement, so we want to find the end of the file as a statement (or chapter) end.
`SourceInfo` says:
```rust
    /// Span of the parser input within the file; all spans reported by the
    /// parser are relative to this.
    pub(crate) span: Span,
```
And indeed in the comment of `Statement::span`, it is explicitly stated that the spans are relative to the segments:
```rust
    /// The textual span of this statement within the segment's buffer.
    ///
    /// Does not include surrounding white space or comments, unlike `span_full()`.
    #[inline]
    #[must_use]
    pub const fn span(&self) -> Span {
        Span::new2(self.statement.label.start, self.span_full().end)
    }
```
So, in order to get the offset of the end of the file, within the last segment, we need to compute the length of that segment, therefore `end - start`.

Here I've been trusting those comments. Are they maybe misleading?

Reply:
All spans are relative to the start of the containing segment. Is this function returning global spans? (This may not even be a well formed notion given that segments can come from different files.) The spec of this function is unclear. If it is a segment-local span, then there should be no need for adjustment.

What I would have expected here is that if there is no next statement because it is the end of the segment, then we should set the end to the end of the segment itself, which I believe is just `segment_span.end` which is what was already here.

Reply:
The function normally returns `statement.span()`s, which are relative to the start of the containing segment.

But `segment_span` is the _global_ span of the segment, so just `segment_span.end` would be the end of the segment relative to the whole file. Here we want the end of the segment relative to the segment itself, therefore `segment_span.end - segment_span.start`.

Reply:
The thing here is that `source_info(stmt.segment.id).span` gives us a _global_ span.